[SVS-89] Update solution binding; use SonarAnalyzer rather than SonarLint packages

mjcheetham opened the pull request:
- Some rule sets are inconsistent
- We're contributing to the continuing download count of the older SonarLint NuGet packages
- We're not bound consistently with the release extension w.r.t. test projects

mjcheetham commented on src/Integration.Vsix/Integration.Vsix.csproj:
```diff
@@ -87,7 +87,9 @@
     <None Include="Integration.Vsix.ruleset" />
-    <None Include="packages.config" />
+    <None Include="packages.config">
+      <SubType>Designer</SubType>
```
No idea why, but VS keeps adding this, even if I remove it.

mjcheetham commented on src/SonarQube/SolutionBinding.sqconfig:
```diff
@@ -1,4 +1,4 @@
 {
-  "ServerUri": "https://dory.sonarsource.com/",
+  "ServerUri": "https://dory.sonarsource.com/sonarqube/",
```
The server URL had changed.

mjcheetham commented on src/TestInfrastructure/TestInfrastructure.csproj:
```diff
@@ -12,6 +12,7 @@
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <TargetFrameworkProfile />
+    <SonarQubeTestProject>true</SonarQubeTestProject>
```
Marking this class library as a test asset.

eyalantzMS commented on src/Integration.UnitTests/Integration.UnitTests.csproj:
```diff
@@ -26,7 +26,6 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <CodeAnalysisRuleSet>Integration.UnitTests.ruleset</CodeAnalysisRuleSet>
```
Can you revert this change please? I prefer to have those than no to have them at all.

eyalantzMS commented on src/Integration.Vsix.UnitTests/Integration.Vsix.UnitTests.csproj:
```diff
@@ -26,7 +26,6 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <CodeAnalysisRuleSet>Integration.Vsix.UnitTests.ruleset</CodeAnalysisRuleSet>
```
Same as above (undo this file)

eyalantzMS commented on src/Integration.Vsix.UnitTests/Integration.Vsix.UnitTests.ruleset:
```diff
@@ -1,4 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<RuleSet Name="SonarQube" ToolsVersion="14.0">
```
And keep this one

eyalantzMS commented on src/Integration.UnitTests/Integration.UnitTests.ruleset:
```diff
@@ -1,11 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
And keep this one

eyalantzMS commented on src/Integration.Vsix.UnitTests/packages.config:
```diff
@@ -4,7 +4,6 @@
   <package id="Microsoft.CodeAnalysis.Common" version="1.0.0" targetFramework="net452" />
   <package id="Microsoft.CodeAnalysis.Workspaces.Common" version="1.0.0" targetFramework="net46" />
   <package id="Microsoft.Composition" version="1.0.27" targetFramework="net46" />
-  <package id="SonarLint" version="1.7.0" targetFramework="net452" developmentDependency="true" />
```
You might want to check with @tvsonar about this bit.

eyalantzMS commented on src/Progress.TestFramework/Progress.TestFramework.csproj:
```diff
@@ -26,7 +26,6 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <CodeAnalysisRuleSet>progress.testframework.ruleset</CodeAnalysisRuleSet>
```
Undo changes

eyalantzMS commented on src/Progress.UnitTests/Progress.UnitTests.csproj:
```diff
@@ -26,7 +26,6 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <CodeAnalysisRuleSet>progress.unittests.ruleset</CodeAnalysisRuleSet>
```
Undo

eyalantzMS commented on src/Progress.UnitTests/Progress.UnitTests.ruleset:
```diff
@@ -1,4 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Undo

eyalantzMS commented on src/TestInfrastructure/TestInfrastructure.ruleset:
```diff
@@ -1,11 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <RuleSet Name="SonarQube" ToolsVersion="14.0">
-  <Include Path="..\sonarqube\sonarlint-visualstudiocsharp.ruleset" Action="Default" />
```
Undo

eyalantzMS commented:
LGTM, other than undoing all the deletion of rulesets from test projects which is better to have (did find a couple of test issues as I recall).

mjcheetham commented:
@eyalantzMS, rather than just revert the changes I've added `<SonarQubeTestProject>false</SonarQubeTestProject>` to all the test projects and updated the binding (and also restored any overrides we had on rules in test project rule sets).

Without these flags we're still going to fall behind on NuGet package updates and other quality profile definition changes for these projects :smiley:

eyalantzMS commented:
It is a fair point, is it applicable to our customers as well?

mjcheetham commented:
Well this is a larger experience hole in both the connected mode and the MSBuild scanner, they're both just straight-up ignoring any test projects, whereas the SonarQube server has special treatment for test projects (and files).

Just discussed this offline with Blair and Duncan. Might be better to test the test projects AS test assets, then just manually 'bind' these projects. Yes they'll fall out of date w.r.t. NuGet package version but they'll still get the updated shared solution-level rule set file.
(Also, when new rules are added we'll be reminded to update the analysers on the test projects as VS will complain about the new rules not existing).

mjcheetham commented:
So basically the idea is we'll be strengthening locally for test projects, but the CI build etc won't take it into account, as before.
